### DISCUSSION TITLE
Cruiser 5.2.0 (Beta)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#690] [Announcements]
## Cruiser 5.2.0 (Beta) is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/570#discussioncomment-11486538
1 beep for left, 2 beeps for right, 3 beeps for other
(Settings | Voice guidance | Advanced)

- https://github.com/devemux86/cruiser/discussions/679
(Settings | Voice guidance | Advanced)

- https://github.com/devemux86/cruiser/discussions/126#discussioncomment-11460963
It works with the internal light / dark map themes.
(long press the direction icon at top left)

- https://github.com/devemux86/cruiser/discussions/694#discussioncomment-11628012
(long press anywhere on the route)

- https://github.com/devemux86/cruiser/discussions/695
(see the link for instructions)

- https://github.com/devemux86/cruiser/discussions/431#discussioncomment-11499867

- **Many improvements**

<img src="https://github.com/user-attachments/assets/05e593e2-7954-4d6d-952c-ccd8b625a9b5" width="50%"/><img src="https://github.com/user-attachments/assets/379034cc-979d-4357-9f2e-346a070a679a" width="50%"/>
<img src="https://github.com/user-attachments/assets/47ccbad4-a338-4527-a824-b0f9fc97e1c4" width="50%"/><img src="https://github.com/user-attachments/assets/e2a7c5bb-e77b-4941-b904-86ef087cc3f9" width="50%"/>

**Comment by devemux86** [comment]
It also contains:

- **Adaptive hill shading that works at all zoom levels**
(reload the map theme)

- https://github.com/devemux86/cruiser/discussions/612 revision 38
(reload the map theme)

**Reply by devemux86** [reply]
Screenshots of the adaptive hill shading can be found here:

- https://github.com/devemux86/cruiser/discussions/612#discussioncomment-11607589
- https://github.com/mapsforge/mapsforge/pull/1592

As you can see, we can now color the hill shading. If there is interest for this, I could add a setting.

**Reply by DolomitesTom** [reply]
Sounds good

**Reply by ToToGSARider** [reply]
Good morning, there has been huge progress in the Cruiser again. How are you getting on with reintroducing 3D and tilting the navigation? Have you already found a solution to the problem? Would it be possible to open a separate article for this topic that you can follow so that you don't always have to write in a general article?


Guten Morgen, es gibt ja wieder riesige Fortschritte im Cruiser. Wie kommt ihr den voran das Thema 3D bzw. das kippen der Navigation wieder einzubauen. Hat man schon Lösungsansätze für das Problem gefunden? Könnte man evtl. für das Thema einen separaten Artikel aufmachen dem man folgen kann, damit man nicht immer in einen allgemeinen Artikel schreiben muss?

**Reply by devemux86** [reply]
No, there is no such plan, as it belongs to [Mapsforge](https://github.com/mapsforge/mapsforge) map library.
If someone implements it, then I can include it in Cruiser as well.

**Reply by ToToGSARider** [reply]
Can't you make a request for Mapsforge? I'm not as technically familiar with it as you are, as I'm only a user. I wouldn't even know where to write, let alone how to describe it.
What does the OsmAnd software use for map material? Is 3D fully available and have I ever read about hillshading in the app? 
As I said, I'm just a user. As already suggested, I don't want to keep asking in the general posts, but would find a separate post useful, which you can then simply keep an eye on. I know that it's not so great for you either if I keep asking in the posts. 

Könnt ihr nicht eine Anfrage nach Mapsforge stellen? Ich kenne mich da technisch nicht so aus wie Ihr, da ich nun einmal nur Anwender bin. Ich wüßte nicht einmal wo ich hinschreiben muß und schon gar nicht, wie ich das beschreiben soll.
Was nutzt den die Software OsmAnd für Kartenmaterial? Da ist 3D komplett verfügbar und Hillshading habe ich da auch schon einmal in der App gelesen? 
Wie gesagt, ich bin nur Anwender. Wie auch schon vorgeschlagen, möchte ich nicht ständig in den allgemeinen Beiträgen nachfragen, sondern fände einen getrennten Beitrag sinnvoll, den man dann einfach im Auge behalten kann. Ich weiss, dass es für euch ja auch nicht so toll ist, wenn ich immer wieder in die Beiträge Frage.

**Reply by devemux86** [reply]
I am the main developer of the [Mapsforge](https://github.com/mapsforge/mapsforge) map library.
It's an open source library, everyone can write and contribute new features. :slightly_smiling_face: 
Like the new [adaptive hill shading](https://github.com/devemux86/cruiser/discussions/612#discussioncomment-11607589).

You can create and monitor a new thread here to request a new feature.

**Comment by devemux86** [comment]
## Cruiser 5.2.0-b2 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/694#discussioncomment-11628012
(long press anywhere on the route)

- https://github.com/devemux86/cruiser/discussions/695
(see the link for instructions)

**Reply by Ciutnik** [reply]
Hi. First of all, best wishes for Christmas, to everyone - those who celebrate and those who don't 🙂

And secondly, I noticed that in 5.2.0-b the selection of alternative routes between via points stopped working. That is, the selection affects the entire route, not just the section between via points. 
It may be strange, but it seems to me that this happened after activating the "Use distant waypoints" function (?).

**Reply by Ciutnik** [reply]
Another thing, although it's BRouter's fault, is the route running through the construction site. The bridge unfortunately does not exist yet, but the Bike profile suggests going this way, despite the OSM tag `highway=construction`. This is probably not the best idea 😉

[through_construction.gpx.txt](https://github.com/user-attachments/files/18247139/through_construction.gpx.txt)

**Reply by devemux86** [reply]
@Ciutnik Thanks for the report and for taking the time to analyze your settings! :+1: 

> it seems to me that this happened after activating the "Use distant waypoints" function (?).

This will work properly in the next update.

Remember that "Use distant waypoints" calculates the whole route.
(like the normal calculation mode)

**Reply by devemux86** [reply]
> Bike profile

@Ciutnik Cruiser bike app profiles use BRouter original trekking routing profile.

If you see anything strange in BRouter original routing profiles,
it is best to report it in [BRouter repository](https://github.com/abrensch/brouter/) with a route example using [BRouter website](https://brouter.de/brouter-web/).

Or even better if you can suggest specific improvements in the BRouter profile.
BRouter profiles can be seen [here](https://github.com/abrensch/brouter/blob/master/misc/profiles2/).

**Reply by Ciutnik** [reply]
Today I noticed something else: the yellow dots at the turnaround points do not show up during active navigation, regardless of their zoom setting. Is this intentional?

And in general, during Christmas bike trips outside the city, the Cruiser works brilliantly. A bit too cold for a motorcycle, but perfect for a bike (I recently got an electric assisted bike... I love it, although I know that "real cyclists" would gladly lynch me 😉).

**Reply by devemux86** [reply]
> Is this intentional?

@Ciutnik Indeed it is, because in navigation we should not auto hide overlays for performance reasons.
The auto hide of overlays now only works in normal map view. In navigation they are either on or off.

**Reply by Ciutnik** [reply]
@devemux86 OK, I understand, although I miss showing them on the route during navigation. Was it always like that?

**Reply by devemux86** [reply]
@Ciutnik They were auto hidden, but this can cause problem in navigation.
Also we cannot add settings for everything, we should decide what is on and off.

**Reply by Ciutnik** [reply]
I don't know what others think, but in my opinion these yellow dots are important information during navigation. They allow us to orient at one glance regarding the several maneuvers awaiting us. Of course, at the appropriate zoom level, according to the setting.

**Reply by Ciutnik** [reply]
When you hear the message "turn right in XX meters," you look at the map and are unable to quickly assess the scale. On a generally winding route, without seeing the yellow dot, you don't know which point the message refers to. That was my observation today.

**Reply by devemux86** [reply]
@Ciutnik We can enable them in navigation.

**Reply by devemux86** [reply]
Other navigators draw large arrows on road that often block the view.
The instruction dots allow better visibility.

**Reply by Ciutnik** [reply]
> Other navigators draw large arrows on road that often block the view.

I agree. Sometimes, in particularly complicated cases (e.g. when there are several maneuvers in close proximity), they can look bizarre, instead of helping, they actually mislead. 

The yellow dot is quite sufficient. After all, we have a precise picture of the route drawn and as we approach the maneuver, the direction is rather obvious 🙂

**Reply by devemux86** [reply]
And we can always use smaller yellow dots if we need to.
After all, they don't contain any numbers.

By the way, early versions of Cruiser, many years ago, had such angular arrows on the road. :slightly_smiling_face:

**Comment by devemux86** [comment]
> early versions of Cruiser, many years ago, had such angular arrows on the road.

Screenshot from 2016:

![navigator](https://github.com/user-attachments/assets/612460c1-e951-4da1-a033-12b6dce6e58d)

**Reply by Ciutnik** [reply]
> Screenshot from 2016:

Cool. But as you say - they cover the route 🙂
This lower, three-part panel is interesting. As you know, OsmAnd allows you to create such groups of panels in rows. For the purposes of cycling, I once used this one with the weather:

<img src="https://github.com/user-attachments/assets/611dedee-7022-4345-8e1d-a540f98db42b" width="50%"/>

**Comment by Ciutnik** [comment]
I have a request to change the translation to Polish.

`Voice guidance | Announcement | Street names | Before the turn`
Is `Przed turą` → it should be `Przed zwrotem`

`Navigation | Panels | Nearby favorites | Appearance`
is `Wygląd` → it should be `Widoczność`
The same applies to POIs.

Thanks in advance 🙂

**Reply by devemux86** [reply]
@Ciutnik Thanks for the translations.

"Przed zwrotem" is translated "Before return"?
We need "Before the turn" not "return". Or it is same in Polish?

Also "Widoczność" is "Visibility", but the English text is "Appearance".

If English need better texts, then we should change them in all languages.

**Reply by devemux86** [reply]
> "Navigation | Panels | Nearby favorites | Appearance"

Should we change this to "Visibility"?

**Reply by Ciutnik** [reply]
> We need "Before the turn" not "return". Or it is same in Polish?

I understand the context - in this case it's about a turn (right/left), not a U-turn. In Polish, a "turn" ("zwrot") can be to the right, left, 180°... It's more or less the same as the universal "maneuver" ("manewr"), which in Polish has a broader meaning, because it also includes, for example, overtaking or passing.

> Also "Widoczność" is "Visibility", but the English text is "Appearance".

In the sense of the panel appearing, the word "Wygląd" makes absolutely no sense and is misleading (it is more about appearance in terms of color, size, etc.). Believe me, "Widoczność" is the most fitting here.

That's how it is with translators. When a given word is translated from one language to another, when you do it back it can come out with a completely different word than it was originally 🙂

**Reply by devemux86** [reply]
@Ciutnik Thanks for the additional details.

I will update the translations, also make it "Visibility" to be common in all languages.

**Reply by Ciutnik** [reply]
> I will update the translations, also make it "Visibility" to be common in all languages

Maybe it's better to wait for the opinion of a native speaker? 😉

**Reply by devemux86** [reply]
For English? Primary language is English.

**Reply by Ciutnik** [reply]
> For English?

Yes. Maybe @MotoUKRider? 🙂

**Reply by MotoUKRider** [reply]
@devemux86 and @Ciutnik I would use "Visibility" when talking about turning something on/off or setting a threshold for it to "appear"(!)

When talking about the style, color etc then "Appearance" is best.

**Reply by Ciutnik** [reply]
Thanks @MotoUKRider and sorry for calling you out on the board... 
Now we have no doubts about how it should be 🙂

**Reply by MotoUKRider** [reply]
@Ciutnik Happy to help!

**Reply by bios4** [reply]
While we're at it...

> "Navigation | Panels | Nearby favorites | Appearance"

The German translation is also not 100% reasonable. I'd like to suggest to change it from "Aussehen" (relates more to colors, text sizes etc.) to "Darstellung" or maybe even better "Anzeige".

Same applies for nearby POI.

**Reply by devemux86** [reply]
@bios4 We'll change it to "Visibility". So "Sichtbarkeit" should be ok?

**Reply by bios4** [reply]
"Sichtbarkeit" will do, yes...

**Comment by devemux86** [comment]
## Cruiser 5.2.0-b3 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- **Map rendering performance improvements**

- https://github.com/devemux86/cruiser/discussions/612 revision 41
(reload the map theme)

**Reply by Ciutnik** [reply]
Thanks for the update! 👍
The only problem I've noticed is that the height diagram isn't displayed in one of the profiles (Bike). Of course it is enabled in the settings. This applies to viewing the map after calculating the route. During navigation, the panel can be displayed correctly. Oddly enough, the almost twin "Mountain Bike" profile don't have this problem.

I've been looking at this for a long time and haven't noticed any rule... I'm posting a problematic profile, maybe you'll notice something strange?

[Rower.json.txt](https://github.com/user-attachments/files/18306413/Rower.json.txt)

**Reply by devemux86** [reply]
@Ciutnik Thanks for the report.
This is strange, I can reproduce it with your profile and will check it.

(if you reset the profile then it should work)

**Reply by devemux86** [reply]
@Ciutnik This will work properly in the next update.

**Reply by Ciutnik** [reply]
@devemux86 Great, thanks!👍 
I did reset the profile, but it's good that the malfunction shouldn't happen again in the future 🙂 

I didn't do anything unusual with the profiles. And if I did anything, it was possibly importing/exporting from the app, in order to share the profile or create a backup.

**Reply by devemux86** [reply]
@Ciutnik What caused the problem was if we changed the order of the top panels in navigation settings.

This will work properly again in the next update.

**Comment by Ciutnik** [comment]
My phone, which I use to navigate while cycling, has an OLED screen. To save it, I activated the screen saver. But it works strangely - the screen turns off after a set time, regardless of whether I use it or not. 

I don't know if it's possible, but I guess each touch should "reset" the countdown until the screen turns off. Now, the timer counts down from when the screen is activated, which sometimes causes it to turn off unexpectedly while viewing the map.

This can be partially prevented by extending the time until the screen turns off, but it does not solve the problem. It should be emphasized that this is not a very troublesome problem, only a little annoying 🙂

**Reply by devemux86** [reply]
@Ciutnik Thanks for the report.

Screen saver works automatically in navigation and follow location modes.
In navigation screen opens at events: next turn instruction, nearby POI or favorite, etc.

You mean that screen saver should reset each time we use the screen.
Just like the auto center / zoom already reset, so we can continue using the screen without interruption.

**Reply by Ciutnik** [reply]
> You mean that screen saver should reset each time we use the screen.

Yes. So that each touch would be an event that resets the time until the screen turns off. 

Indeed, the countdown to centering or restoring zoom works similarly.

**Reply by devemux86** [reply]
> Yes. So that each touch would be an event that resets the time until the screen turns off.

@Ciutnik This will be available in the next update.

**Comment by DolomitesTom** [comment]
I couldn't install 5.2.0 b3, just a message "was not installed".

**Reply by devemux86** [reply]
@DolomitesTom This Beta is closed.

**Cruiser 5.2.0** is on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser):
- https://github.com/devemux86/cruiser/discussions/701

**Reply by DolomitesTom** [reply]
ah ok thx
